### PR TITLE
update ja/development/testing.rst

### PR DESCRIPTION
**Pull request by yama**
Since there were many omissions in translation of the latest document, it was reflected.
I avoided direct translation and changed it to a natural Japanese expression.

thanks.

### CODE REVIEW

File: ja/development/testing.rst [hunk 1]
```diff
@@ -414,14 +407,95 @@ PHPUnitの設定
...
+※リスナーは非推奨であり、:doc:`フィクスチャ構成を更新</appendices/fixture-upgrade>`_する必要があります。
+
+.. _creating-test-database-schema:
+
+テスト用のデータベーススキーマ作成
+----------------------------
+
+CakePHPのマイグレーション機能・SQLダンプファイルのロード、または他のスキーマ管理ツールを使用して、テスト用のデータベーススキーマを生成できます。アプリケーションの ``tests/bootstrap.php`` ファイルにスキーマを作成する必要があります。
+
+CakePHPの`マイグレーションプラグイン </migrations>`を使用してアプリケーションのスキーマを管理する場合は、それらのマイグレーションを利用してテストデータベーススキーマを生成することもできます。
```
Comment (markstory) [inline ja/development/testing.rst]:
```suggestion
CakePHPの`マイグレーションプラグイン </migrations>`を使用してアプリケーションのスキーマを管理する場合は、それらのマイグレーションを利用してテストデータベーススキーマを生成することもできます。::
```

This needs to end in `::` to make the next section a code block.

File: ja/development/testing.rst [hunk 2]
```diff
@@ -414,14 +407,95 @@ PHPUnitの設定
...
+    $migrator->run();
+
+    // Run migrations for multiple plugins
+    $migrator->run(['plugin' => 'Contacts']);
+
+    // Run the Documents migrations on the test_docs connection.
+    $migrator->run(['plugin' => 'Documents', 'connection' => 'test_docs']);
+
+
+複数のマイグレーションを実行する必要がある場合は、次のように実行できます。
```
Comment (markstory) [inline ja/development/testing.rst]:
```suggestion
複数のマイグレーションを実行する必要がある場合は、次のように実行できます。::
```

File: ja/development/testing.rst [hunk 3]
```diff
@@ -414,14 +407,95 @@ PHPUnitの設定
...
+        ['plugin' => 'Documents', 'connection' => 'test_docs']
+    ]);
+
+``runMany()`` を使うと、データベースを共有するプラグインが、各マイグレーションが実行される時にテーブルをドロップしないようになります。
+
+マイグレーションプラグインは、適用されていないマイグレーションのみを実行し、カレントのマイグレーションヘッドが適用されたマイグレーションと異なる場合はマイグレーションをリセットします。
+
+データソース構成のテストでマイグレーションを実行する方法を構成することもできます。詳細については、:doc:`マイグレーションに関するドキュメント </migrations>`を参照してください。
+
+SQLダンプファイルをロードしたい場合は、下記のメソッドを使用できます。
```
Comment (markstory) [inline ja/development/testing.rst]:
```suggestion
SQLダンプファイルをロードしたい場合は、下記のメソッドを使用できます。::
```

File: ja/development/testing.rst [hunk 4]
```diff
@@ -414,14 +407,95 @@ PHPUnitの設定
...
+
+各テスト実行の ``SchemaLoader`` 開始時に、コネクションに紐づく全のテーブルを削除し、提供されたスキーマファイルに基づいてテーブルを再構築します。
+
+.. versionadded:: 4.3.0
+    SchemaLoaderが追加されました。
+
+.. _fixture-state-management:
+
+フィクスチャステートマネージャ
+----------------------
```
Comment (markstory) [inline ja/development/testing.rst]:
```suggestion
フィクスチャステートマネージャ
--------------------------
```

Titles need to have underlines equal to the visual length of the heading.

File: ja/development/testing.rst [hunk 5]
```diff
@@ -414,14 +407,95 @@ PHPUnitの設定
...
+
+デフォルトでは、CakePHPは、データベース内のすべてのテーブルを truncate することにより、各テストの最後にフィクスチャの状態をリセットします。この処理は、アプリケーションが大きくなるにつれてコストがかかる可能性があります。``TransactionStrategy`` を各テストメソッドに使用すると、テストの最後にロールバックされるトランザクション内で実行されます。これによりパフォーマンスが向上しますが、各テストの前に自動インクリメント値がリセットされないため、テストで静的フィクスチャデータに大きく依存しないようにする必要があります。
+
+フィクスチャの状態管理は、テストケース内で定義できます。
```
Comment (markstory) [inline ja/development/testing.rst]:
```suggestion
フィクスチャの状態管理は、テストケース内で定義できます。::
```

File: ja/development/testing.rst [hunk 6]
```diff
@@ -562,150 +579,112 @@ modified のタイムスタンプに今日の日付を反映させたいので
...
-テストケース内のフィクスチャーのロード
---------------------------------------
+上記の例では、各フィクスチャが ``tests/Fixture/Blog/`` ディレクトリからロードされます。
 
-フィクスチャーを作成したらそれらをテストで使いたくなることでしょう。
-各テストケースではあなたが必要としているフィクスチャーをロードすることができます。
-クエリーの実行に際して必要となるモデルのフィクスチャーをロードする必要があります。
-フィクスチャーをロードするには、テストケースに ``$fixtures`` プロパティーを設定します。 ::
+フィクスチャファクトリー
+-----------------
```
Comment (markstory) [inline ja/development/testing.rst]:
```suggestion
フィクスチャファクトリー
---------------------
```

File: ja/development/testing.rst [hunk 7]
```diff
@@ -562,150 +579,112 @@ modified のタイムスタンプに今日の日付を反映させたいので
...
-        {
-            $this->loadFixtures('Articles', 'Comments');
-        }
-    }
+ファクトリーは、関連するフィクスチャの生成にも役立ちます。記事が複数の著者に属していると仮定すると、たとえば、それぞれ5つの記事を持つ2人の著者を作成できます。
 
-あなたはサブディレクトリーにフィクスチャーをロードすることができます。複数ディレクトリーを使用することは、
-大規模なアプリケーションで、フィクスチャーを整理しやすくします。サブディレクトリー中のフィクスチャーを
-ロードするためには、フィクスチャー名にサブディレクトリーを加えてください。 ::
+    $articles = ArticleFactory::make(5)->with('Authors', 2)->getEntities();``
```
Comment (markstory) [inline ja/development/testing.rst]:
```suggestion
    $articles = ArticleFactory::make(5)->with('Authors', 2)->getEntities();
```

File: ja/development/testing.rst [hunk 8]
```diff
@@ -562,150 +579,112 @@ modified のタイムスタンプに今日の日付を反映させたいので
...
     }
 
-上記の例では、両方のフィクスチャーは ``tests/Fixture/Blog`` からロードされることになります。
+このメソッドは、``Application`` インスタンスの作成と、そのインスタンスでの ``routes()`` メソッドの呼び出しを行ないます。
+この ``Application`` インスタンスのコンストラクタには、 ``loadRoutes($constructorArgs)`` としてパラメータを渡すことができます。
+
+プラグインをロード
+------------------------
+
+プラグインをロードしたい場合は ``loadPlugins()`` メソッドを使用できます。
```
Comment (markstory) [inline ja/development/testing.rst]:
```suggestion
プラグインをロードしたい場合は ``loadPlugins()`` メソッドを使用できます。::
```

**Review by markstory — COMMENTED** [review]
Looking great, there are a few formatting changes that need to be made before this can be merged.

**Comment by yama** [comment]
very thanks!

**Comment by othercorey** [comment]
Please check the errors.

**Comment by yama** [comment]
https://stackoverflow.com/questions/68603006/build-process-in-github-actions-completed-with-exit-code-2
GitHub Action configuration issue?

**Comment by yama** [comment]
image
ok, I will fix it later.

**Comment by othercorey** [comment]
Just need to fix some titles and spaces :)

**Comment by yama** [comment]
@othercorey @markstory 

Completed. thanks

**Comment by othercorey** [comment]
Thanks!

**Pull request merged**